private/buf/buflsp/symbol.go

```diff
@@ -523,6 +511,41 @@ func (s *symbol) FormatDocs(ctx context.Context) string {
 	return tooltip.String()
 }
 
+// formatComment takes a raw comment string and formats it by removing comment
+// delimiters and unnecessary whitespace. It handles both single-line (//) and
+// multi-line (/* */) comments.
+func formatComment(comment string) string {
+	comment = strings.TrimSpace(comment)
+
+	if strings.HasPrefix(comment, "//") {
+		// NOTE: We do not trim the space here, because indentation is
+		// significant for Markdown code fences, and if every line
+		// starts with a space, Markdown will trim it for us, even off
+		// of code blocks.
+		return strings.TrimPrefix(comment, "//")
+	}
+
+	if strings.HasPrefix(comment, "/*") && strings.HasSuffix(comment, "*/") {
+		comment = strings.Trim(comment, "/*")
+		// single-line
+		if !strings.Contains(comment, "\n") {
+			return strings.TrimSpace(comment)
+		}
+
+		lines := strings.Split(strings.TrimSpace(comment), "\n")
+		for i, line := range lines {
+			line = strings.TrimSpace(line)
+			line = strings.TrimLeft(line, "*")
+			line = strings.TrimPrefix(line, " ")
+			lines[i] = line
+		}
```

Comment:
I would prefer if this only triggered if the comment starts with `/**` Otherwise, the following will be mangled:

```proto
/*

my things:
* foo
* bar
* baz

*/
```

Reply:
I would also like to see a comment that explains that what this is processing is a Doxygen-style comment, with an inline example of what we expect the comment to look like.